components/l10n/common/locale_util.cc

```diff
@@ -5,78 +5,114 @@
 
 #include "brave/components/l10n/common/locale_util.h"
 
-#include <algorithm>
-#include <vector>
-
-#include "base/strings/string_split.h"
-#include "base/strings/string_util.h"
-#include "base/strings/utf_string_conversions.h"
-#include "ui/base/resource/resource_bundle.h"
+#include "absl/types/optional.h"
+#include "base/check.h"
+#include "base/no_destructor.h"
+#include "brave/components/l10n/common/locale_constants.h"
+#include "servloc.h"
+#include "third_party/icu/source/common/unicode/locid.h"
+#include "ui/base/l10n/l10n_util.h"
 
 namespace brave_l10n {
 
 namespace {
 
-const char kDefaultLanguageCode[] = "en";
-const char kDefaultCountryCode[] = "US";
+absl::optional<icu::Locale>& ICULocaleForTesting() {
+  static base::NoDestructor<absl::optional<icu::Locale>> icu_locale;
+  return *icu_locale;
+}
 
-}  // namespace
+absl::optional<icu::Locale>& LastICULocaleForTesting() {
+  static base::NoDestructor<absl::optional<icu::Locale>> last_icu_locale;
+  return *last_icu_locale;
+}
 
-std::string GetLanguageCode(
-    const std::string& locale) {
-  const std::vector<std::string> locale_components = base::SplitString(locale,
-      ".", base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+inline icu::Locale GetNormalizedICULocale(const std::string& locale) {
+  return {l10n_util::NormalizeLocale(locale).c_str()};
+}
 
-  if (locale_components.empty()) {
-    return kDefaultLanguageCode;
+inline std::string GetLocale(const icu::Locale& icu_locale) {
```

Comment:
GetLocaleString ?

Reply:
but why aren't we using `GetApplicationLocale` here? It seems like we're duplicating functionality in ui/base/l10n here, but with an hard-coded default locale.

Reply:
Discussed via DM so closing comment as pushing new changes, but we will still need to agree a name.